testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ServiceTest.scala

```diff
@@ -152,14 +152,17 @@ object ServiceTest {
       result match {
         case asyncResult: Future[_] =>
           import testServer.executionContext
-          asyncResult.onComplete { _ =>
-            testServer.stop()
-          }
+          // whether the future `asyncResult` was successful or failed, stop the server.
+          asyncResult.andThen {
+            case theResult => {
+              testServer.stop()
+              theResult
+            }
+          }.asInstanceOf[R]
```

Comment:
Great... didn't know about `andThen`. Seems perfect. Reviewing the docs, I don't think it's even necessary to return `theResult`. The partial function is only used for its side effect.

> The andThen combinator is used purely for side-effecting purposes. It returns a new future with exactly the same result as the current future, regardless of whether the current future failed or not. 

(From [Futures and Promises](http://docs.scala-lang.org/overviews/core/futures.html))

So, I think this could be simplified to

```scala
asyncResult.andThen { case _ => testServer.stop() }.asInstanceOf[R]
```